### PR TITLE
fix the relocalized-url tag, and update some more templates to use it

### PR DESCRIPTION
**Pull request by Pomax**
Closes https://github.com/mozilla/foundation.mozilla.org/issues/6965 while also fixing up the `relocalized_url` template tag necessary for resolving that issue.

Closes https://github.com/mozilla/foundation.mozilla.org/issues/7233 because of the widened scope.

The STR will require grabbing a copy of the most recent production data, so that you have real localized content to test on.

- when on `/de/` the http://localhost:8000/blog index page should link to `/de/` links for all cards rather than `/en/`
- same for the http://localhost:8000/campaigns index page.
- the PNI homepage should have categories and products that all link to the correct locale (`en` on EN, `de` on DE, etc)
- the PNI "all" category and logo should point to the homepage on the current locale
- "related products" for PNI products should have URLs for the current locale
- the homepage should also point to `/de/` things when on http://localhost:8000/de/ etc.

Also note that the blog card has had the code for "showing tags" removed, since we haven't exposed that in years and we're already working on taging tags out (for UI purposes).

### CODE REVIEW

**Review by TheoChevalier — CHANGES_REQUESTED** [review]
Working well overall, just spotted a couple areas we will need to fix, that don’t seem hard to fix (?)
Also tested the Blog set component, blog titles have the localized URL, but categories don’t. 

Capture d’écran 2021-08-19 à 17 08 22

Capture d’écran 2021-08-19 à 17 53 25


Capture d’écran 2021-08-19 à 17 57 27

**Review by TheoChevalier — APPROVED** [review]
Actually, the category URL was fixed in the other PR, and I just noticed the faker title issue happens on another PR, so it’s definitely not caused by this code

**Comment by Pomax** [comment]
I don't know why, but this PR causes a redirect test that passes in `master` to fail. I'll have to dig into why.

**Comment by Pomax** [comment]
So https://github.com/mozilla/foundation.mozilla.org/pull/7236/files#diff-2a95e5e43636ba7a64323d87d452a5a27c79dad85ed2daab647f6c0f0105826cR59-R61 is breaking redirects, somehow, and I've love to dig into the whole "why".

**Comment by Pomax** [comment]
Right, found the problem here: `privacynotincluded/products/blah` would fail because the products.py code was directly referencing the `relocalized_url` template tag, instead of calling the "non-template-tag-code" that should have backed `relocalized_url`, so e31c7a1bfce8bfb1c25cff03b9e6d8adf206cff1 fixes that.

File: network-api/networkapi/wagtailpages/templatetags/localization.py [hunk 1]
```diff
@@ -53,19 +54,26 @@ def get_unlocalized_url(page, locale):
...
 
+# Force-relocalize a URL
+@register.simple_tag(takes_context=True)
+def relocalized_url(context, url, locale_code=None):
+    if locale_code is None:
+        request = context['request']
+        locale_code = get_language_from_request(request, True)
+    return relocalize_url(url, locale_code)
+
+
```
Comment (Pomax) [inline network-api/networkapi/wagtailpages/templatetags/localization.py:71]:
```suggestion
# Force-relocalize a URL
@register.simple_tag(takes_context=True)
def relocalized_url(context, url):
    request = context['request']
    locale_code = get_language_from_request(request, True)
    return relocalize_url(url, locale_code)


```

**Comment by Pomax** [comment]
@TheoChevalier @KalobTaulien one last pass: tests pass now, and I'm pretty sure we're good to go now, but just in case I had a typo in that last commit =D

**Review by TheoChevalier — APPROVED** [review]
Thanks for digging that! Looking good

**Pull request merged**